src/check/runner/configuration/GlobalParameters.ts

```diff
@@ -4,8 +4,19 @@ import { Parameters } from './Parameters';
 /** @internal */
 const globalParametersSymbol = Symbol.for('fast-check/GlobalParameters');
 
+type HookFunction = () => void;
+type AsyncHookFunction = (() => Promise<unknown>) | (() => void);
+
+type ExtraGlobalParameters = {
+  beforeEach?: HookFunction;
+  afterEach?: HookFunction;
+  asyncBeforeEach?: AsyncHookFunction;
+  asyncAfterEach?: AsyncHookFunction;
+};
```

Comment:
Not sure I understand 1 :thinking: 
What's the problem with being defined together?

Reply:
In reality, the existence of both `property` (synchronous) and `asyncProperty` (asynchronous) is there for legacy reasons. Ideally all properties should be defined as asynchronous.

For instance, if you take a look into [this wrapper around fast-check designed for ava](https://github.com/dubzzz/ava-fast-check/), there is no more distinction between synchronous and asynchronous properties. Behind the scene everything is asynchronous as it only uses asyncProperty (but users don't know).

Reply:
If you define the two global hooks together, there is probably an issue in your test suite. Or maybe hooks should not have been defined at a global level but at file or directory level.

My point is:
- If no global: property/asyncProperty take what the user specify using the beforeEach/afterEach methods
- If only global beforeEach: property/asyncProperty take it but replace it if the user specify one using the beforeEach method
- If only global asyncBeforeEach: property throws / asyncProperty takes it but replaces it if the user specify one using the beforeEach method
- If global beforeEach and asyncBeforeEach: property/asyncProperty throw

Reply:
As mentioned in https://github.com/dubzzz/fast-check/pull/900#discussion_r473277266, I think that it is valuable to not replace the global hooks when new ones have been defined with the beforeEach/afterEach methods.
If I have this case:
```typescript
configureGlobal({
    beforeEach: () => console.log('cleanup'),
});

fc.property(() => {}).beforeEach(() => console.log('beforeEach'));
```
I want both hooks to be run. Otherwise I'm going to have to call `cleanup` every time I use the `beforeEach` method in a property/asyncProperty.
Although I understand your point, an API like the one you suggest wouldn't solve my problem unfortunately :sweat_smile: